### ISSUE TITLE
enhancement: Support Refresh-DeploymentClients

### ISSUE THREAD

**Issue by adrianhall** [splunk/splunk-reskit-powershell#3]
Currently, we have pretty good support for the deployment server.  However, we cannot do the "splunk refresh deploy-clients" equivalent through a command.  The REST endpoint is deployment/server/{name}/check_new.

Please add a PowerShell command to refresh the deployment clients.  It should support a named client or a list of clients or all clients.

Thanks,

**Comment by halr9000** [comment]
Looks like the best verb would be either Sync or Update.
http://msdn.microsoft.com/en-us/library/windows/desktop/ms714428(v=vs.85).aspx.
I'm leaning towards update.

Note that you can have a refresh alias if you really think it's needed. I'd
suggest not, but perhaps a Reload alias might be good to catch those
familiar with the Splunk CLI command.
On Aug 1, 2012 9:25 AM, "Adrian Hall" <
reply@reply.github.com>
wrote:

**Comment by adrianhall** [comment]
Update-DeployClients would be good.  I can put in the refresh alias if needed (which is the splunk nomenclature).

Reload would not be appropriate, since we aren't reloading anything.

**Comment by halr9000** [comment]
Oh! I saw you type "refresh" but parsed it as "reload" in my head. I didn't even know there were two commands for this. I've always done "splunk reload deploy-server" to...refresh the clients. I've perhaps been using the wrong command all along, need to research that.

**Comment by adrianhall** [comment]
Yes, you have.  splunk reload deploy-server reloads the serverclass.conf file.  splunk refresh deploy-clients tells the clients that the apps have changed.
